sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -82,34 +89,67 @@
 public class CoderRegistry implements CoderProvider {
 
   private static final Logger LOG = LoggerFactory.getLogger(CoderRegistry.class);
+  private static final Map<Class<?>, CoderFactory> REGISTERED_CODER_FACTORIES_PER_CLASS;
+
+  static {
+    Set<CoderRegistrar> registrars = Sets.newTreeSet(ObjectsClassComparator.INSTANCE);
+    registrars.addAll(Lists.newArrayList(
+        ServiceLoader.load(CoderRegistrar.class, ReflectHelpers.findClassLoader())));
+    Multimap<Class<?>, CoderFactory> codersToRegister = HashMultimap.create();
+    codersToRegister.put(Byte.class, CoderFactories.fromStaticMethods(ByteCoder.class));
```

Comment:
I wanted to ensure that the "standard" coders were always chosen as the defaults since they are added to the multimap first. By putting it into a registrar, they may not be the defaults.

Once we have the standard coders split from the custom coders, we can move the custom coders out to a registrar.